app/src/main/java/org/mozilla/fenix/library/history/HistoryAdapter.kt

```diff
@@ -60,11 +99,31 @@ class HistoryAdapter(
             timeGroup = current.historyTimeGroup
         }
 
-        holder.bind(current, timeGroup, position == 0, mode, isPendingDeletion)
+        // If there is a single visible item, it's enough to change the empty state of the view.
+        if (isEmpty && !isPendingDeletion) {
+            isEmpty = false
+            onEmptyStateChanged.invoke(isEmpty)
+        } else if (position + 1 == itemCount) {
+            // If we reached the bottom of the list and there still has been zero visible items,
+            // we can can change the History view state to empty.
+            if (isEmpty) {
+                onEmptyStateChanged.invoke(isEmpty)
+            }
+        }
+
+        holder.bind(
+            current,
+            timeGroup,
+            position == 0,
+            mode,
+            isPendingDeletion,
+            groupPendingDeletionCount
+        )
     }
 
-    fun updatePendingDeletionIds(pendingDeletionIds: Set<Long>) {
-        this.pendingDeletionIds = pendingDeletionIds
+    @Suppress("UndocumentedPublicFunction")
```

Comment:
Is there any specific reason why we are suppressing ? :)

Reply:
It looks like Documentation should add something meaningful, which is not immediately obvious from the code. I can't come up with anything about a
```
fun updatePendingDeletionItems(pendingDeletionItems: Set<PendingDeletionHistory>) {
        this.pendingDeletionItems = pendingDeletionItems
    }
```
really :(

Reply:
added documentation